### PR TITLE
Implement wrappers with less generated code

### PR DESCRIPTION
**Pull request by hexaglow**
Add `wrap.ServerToClient`, which converts any gRPC server implementation to a `grpc.ClientConnInterface`.
This can be passed to the appropriate gRPC client constructor,
so any server implementation can be trivially adapted into a client.

Reimplement `protoc-gen-wrapper` to use `wrap.ServerToClient`.
The generated code is reduced to a near-identical wrapper type for each service, without generated code for every service method.

Note: regenerated wrappers aren't in a commit yet. I'll add them once the review of the handwritten code is complete.

### CODE REVIEW

File: pkg/wrap/wrap.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+// srv is the gRPC server implementation type (a type that could be passed to a grpc RegisterXxxServer function).
+//
+// Unary and streaming calls are supported.
+//
+// The call options grpc.Header and grpc.Trailer are supported for unary RPCs. All other call options are ignored.
+func ServerToClient(desc grpc.ServiceDesc, srv any) grpc.ClientConnInterface {
+	// check that srv is the right type to be a server for desc
+	expectType := reflect.TypeOf(desc.HandlerType).Elem()
+	if !reflect.TypeOf(srv).Implements(expectType) {
+		panic(fmt.Sprintf("grpcdynamic: srv must be of type %v", expectType))
```
Comment (mattnathan) [inline pkg/wrap/wrap.go:24]: Where does this panic prefix come from? Shouldn't it be `wrap` or something?

File: pkg/wrap/wrap_test.go
Comment (mattnathan) [inline pkg/wrap/wrap_test.go:1]: Can you add some tests for when the server responds with an error please. I know the old implementation had some issues around streams ending by returning an error.

**Review by mattnathan — CHANGES_REQUESTED** [review]
The code looks excellent. As I said on the call, good find with this approach, much more maintainable.

File: pkg/wrap/wrap.go (hunk 1 above)
Comment (hexaglow) [reply pkg/wrap/wrap.go:24]: It's a leftover from being moved. It should indeed be `wrap`

**Pull request merged**